templates/auth0-singlelogout-handler.php

```diff
@@ -6,7 +6,7 @@
 
 $logout_url = wp_logout_url();
 $logout_url = html_entity_decode( $logout_url );
-$logout_url = add_query_arg( 'redirect_to', get_permalink(), $logout_url );
+$logout_url = add_query_arg( 'redirect_to', ( get_the_ID() ? get_permalink() : wp_login_url() ), $logout_url );
```

Comment:
please explain the logic behind this line

Reply:
`get_the_ID()` checks if we're on a page/post, which means `get_permalink()` will return something useful. If not, we send them to the login page where Lock will appear.